Logic/Wechat.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace jean553\WechatBundle\Logic;
+
+class Wechat
+{
+    const WECHAT_AUTH2_URL =
+        "https://api.weixin.qq.com/sns/oauth2/access_token?";
+
+    /**
+     * @var string $openId Wechat openid of the user
+     */
+    private $openid;
+
+    /**
+     * Authenticate the user, check if the connection is
+     * done through the WeChat browser. Use the appid
+     * and secret passphrase to authenticate the user.
+     *
+     * @param string $appid WeChat application identifier
+     * @param string $secret WeChat application secret
+     *
+     * @return boolean authentication succeed or failure
+     */
+    public function authenticate($appid, $secret) {
```

Comment:
you're missing the third parameter then, 
I thought  you were using authenticate to  identify YOUR SERVER , against wechat, not the current current

so the wechat code needs to be given in parameter